packages/neo-one-monitor/src/BrowserMonitor.js

```diff
@@ -30,30 +55,193 @@ class BrowserGauge extends BaseMetric implements Gauge {
 }
 
 class BrowserHistogram extends BaseMetric implements Histogram {
-  // eslint-disable-next-line
-  observe(countOrLabels?: number | RawLabels, count?: number): void {}
+  observe(countOrLabels?: number | RawLabels, count?: number): void {
+    this.values.push({ countOrLabels, count });
+  }
 }
 
 class BrowserSummary extends BaseMetric implements Summary {
   // eslint-disable-next-line
   observe(countOrLabels?: number | RawLabels, count?: number): void {}
 }
 
-class BrowserMetricsFactory implements MetricsFactory {
+interface MetricsFactoryCollect extends MetricsFactory {
+  collect(): Object;
+}
+
+type MetricCollection = {|
+  counters: Array<BaseMetric>,
+  gauges: Array<BaseMetric>,
+  histograms: Array<BaseMetric>,
+  summaries: Array<BaseMetric>,
+|};
+
+class BrowserMetricsFactory implements MetricsFactoryCollect {
+  _metrics: MetricCollection;
+
+  constructor() {
+    this._metrics = {
+      counters: [],
+      gauges: [],
+      histograms: [],
+      summaries: [],
+    };
+  }
+
+  collect(): MetricCollection {
+    const metrics = JSON.parse(JSON.stringify(this._metrics));
```

Comment:
This is a janky pattern. Makes a lot of assumptions about how `BaseMetric` is implemented. Instead, you should add an explicit `serializeJSON` method.